### PR TITLE
[Form] rework form validator tests

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

This will allow writing tests for #37103.

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    public function testCascadeValidationToChildFormsUsingPropertyPathsValidatedInSequence()
+    {
+        $form = $this->formFactory->create(FormType::class, null, [
+            'validation_groups' => new GroupSequence(['group1', 'group2']),
+        ])
+            ->add('field1', null, [
+                'constraints' => [new NotBlank(['groups' => 'group1'])],
+                'property_path' => '[foo]',
+            ])
+            ->add('field2', null, [
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php:145]: While working on #37085 I realised that we couldn't add this second field and assert its constraints are not validated. Investigating the cause for it I found out that the default `ConstraintValidatorTestCase` is not well suited for these kinds of tests where we cascade validation to nested fields.

**Comment by xabbuh** [comment]
ping @symfony/mergers This is blocking #37103. :)

**Comment by yceruto** [comment]
This looks good to me, but Travis isn't running here, does anyone know why?

**Comment by yceruto** [comment]
Well, it looks like yes https://travis-ci.org/github/symfony/symfony/builds/696496883, but my github UI isn't displaying it :confused:

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request closed without merging**